### DISCUSSION TITLE
Enable logging for errors

### DISCUSSION THREAD

**Discussion by nihiluis** [TanStack/query#7198] [Q&A]
Hi, I'm a fresh user of this lib and got my first queries running. If an error occurs, nothing is printed into the console.
The docs seem to indicate that it should log to console on development mode. I have the dev tools added and process.env.NODE_ENV = "development" (through Vite)

Do I need to do anything else?

**Comment by TkDodo** [comment]
> If an error occurs, nothing is printed into the console.

not sure which errors you are talking about. No, we are not logging rejected Promises returned from the `queryFn` to the console. We used to do that in the past but it was confusing because you had to turn it off for testing.

**Reply by nihiluis** [reply]
Ok, thanks. What's a good way to enable it? It's sort of confusing to me as I currently don't have away to notice errors in my queries and mutations.

**Reply by TkDodo** [reply]
not sure I understand the question. Your query will be in `error` state if you get an error, so it would be good to handle that. You can also integrate with error boundaries. There is an [`onError` callback](https://tanstack.com/query/v5/docs/reference/QueryCache) on the `queryCache` that you can also use.

**Reply by Yihao-G** [reply]
`QueryCache#onError` is good for logging `queryFn`'s errors globally, but it won't be called if anything else (specifically the `select` function) errors.

Is there a React Query way to log errors for the `select` functions globally too?

**Reply by TkDodo** [reply]
not that I know of
